OPHD/States/MapViewStateIO.cpp

```diff
@@ -393,6 +393,24 @@ void MapViewState::readStructures(Xml::XmlElement* element)
 			mineFacility.mine(mine);
 			mineFacility.maxDepth(mTileMap->maxDepth());
 			mineFacility.extensionComplete().connect(this, &MapViewState::onMineFacilityExtend);
+
+			auto trucks = structureNode->firstChildElement("trucks");
+			if (trucks == nullptr)
+			{
+				throw std::runtime_error("MapViewState::readStructures(): MineFacility structure saved without a trucks node.");
+			}
```

Comment:
Just wanted to comment that a check like this will invalidate all saved games where a mine facility exists.

Reply:
True. Maybe I should change this to try to read data if it's not `nullptr` versus throwing if it's not found? It's not really a change that needs to throw away older saves.

Reply:
That's kind of what I was thinking. Could potentially just accept any lost data, or perhaps restore using default values.